worker-core/src/main/java/com/hpe/caf/worker/core/WorkerCore.java

```diff
@@ -549,7 +556,7 @@ public void forward(TaskInformation taskInformation, String queue, TaskMessage f
                     workerQueue.acknowledgeTask(taskInformation);
                 } else {
                     // Else forward the task
-                    byte[] output = codec.serialise(forwardedMessage);
+                    final byte[] output = convertAndSerializeMessage(forwardedMessage, publishTaskDataAsObj);
```

Comment:
Ideally, it is required for pause(or anything else which sends message to the pipes). I had checked with @rorytorneymf though and found that pause/unpause is on hold at the moment. So, thought of not making any changes since a corresponding change is required for unpause which is not merged yet. 
But now it seems harmless even if the change is done for pause. So will take care of it while implementing one of the other approaches brought out by you for handling this change.

Reply:
Addressed this for pause as well in 8c8dafd. Resolving other comments where same thing is pointed out.